Added current games to static view

JefeNelson opened the pull request:
Modified the available scope to view the current games available

JefeNelson committed:
Added current games to static view

smiths15 commented on app/models/game.rb:
```diff
@@ -1,6 +1,7 @@
 class Game < ApplicationRecord
 
   scope :available, -> { where(black_player_id: nil) }
+  scope :current, -> { where(black_player_id: !nil) }
```
@JefeNelson It looks good, the only thing is that when I created a game, I didn't see my game under current game. Did it work for you? 

@xelzi we may need to also revisit the available game scope and add conditions in the index view so that only a user who didn't create a game and has the option to join should see games available

JefeNelson committed:
Fixed current games on static view and set pieces on games controller

JefeNelson committed:
Update index.html.erb

JefeNelson committed:
Merge branch 'master' into more_scope

JefeNelson committed:
Fixed error in the static controller's index method

JefeNelson committed:
Fixed error in the static controller's index method

JefeNelson committed:
Removed set pieces and set default turn call in game model

JefeNelson commented:
- Changed the current games view to show the games only to the users who are in them (static index)
- Changed the games show view to show join button to only users who are not in them. (games show)
- Changed the games controller to call set_pieces and set_default_turn. Calling them in the Game.create method. (removed calls from game model)

JefeNelson commented:
To test these changes 
- Create two users. Log one in and log another in incognito mode.
- Create a game with one. There shouldn't be a join button. It should show on the home page under available and current.
- Join the same game with the other. Refresh both, and the game should only show under current.
- For further testing, create/log in a third user and see if the game shows up under available or current. (It shouldn't)

smiths15 commented:
Started new game with User1, Game1 under current and available games and didn't see Game1 join button :thumbsup:
Created User2 and saw Game1 join button under available games :thumbsup: 
Joined Game1 with User2 and saw Game1 under current games :thumbsup: 
Created User3 and Game2 and it appeared in available games for User1 :thumbsup:

smiths15 commented on app/views/static/index.html.erb:
```diff
@@ -13,8 +13,18 @@
-<br>
+<% end %>
 
+<h1> Current Games </h1>
```
Small suggestion: I was thinking that as a user, when I log in I would want to see all my current games before seeing available games. Other than that everything looks good!

Edwin-F replied:
I agree with for this future change. Other than that, it works.

smiths15 commented on app/controllers/games_controller.rb:
```diff
@@ -11,6 +11,8 @@ def new
 
   def create
     @game = Game.create(white_player_id: current_user.id)
+    @game.set_pieces_on_board
```
works great!

Edwin-F commented on app/views/games/show.html.erb:
```diff
@@ -3,7 +3,8 @@
 <br>
 
```
It might be nice in the future to add html that tells the user what player they are: either black or white, since the board orientation is the same.

Edwin-F reviewed (approved):
Overall, it works. In the future, maybe we can incorporate rspec test for the games_controller to update when user2 joins a game, at least to test the functionality.
